### PR TITLE
Update ocm client and schemas

### PR DESCRIPTION
**Pull request by ashishmax31**
# py-mtcli

## Description

Short description of the change:

- [Schema change]Renamed `subscriptionConfig` to `config`.
- [Schema change]Moved the `secrets` attribute under the now renamed `config` attribute
- Addon Loader updates to include methods for accessing attributes.
- Update the ocm client to include secretPropagations attribute in the addon/addon-version payload.

### CODE REVIEW

File: managedtenants/data/selectorsyncset.yaml.j2 [hunk 1]
```diff
@@ -155,14 +155,14 @@ items:
...
-          - name: {{env_obj['name']}}
-            value: |-
-              {{env_obj['value']}}
-          {% endfor %}
+            env:
+            {% for env_obj in ADDON.get_envs() %}
+            - name: {{env_obj['name']}}
+              value: |-
+                {{env_obj['value']}}
+            {% endfor %}
```
Comment (rpriyanshu9) [inline managedtenants/data/selectorsyncset.yaml.j2:165]: Hey @ashishmax31! Can you confirm the `indentation` here is correct?

Comment (ashishmax31) [reply managedtenants/data/selectorsyncset.yaml.j2]: Thanks, fixed.

**Review by rpriyanshu9 — APPROVED** [review]
LGTM

**Comment by ashishmax31** [comment]
> LGTM. Just one thing, now that the `secrets` are present inside the `config` attribute, won't the existing Addons fail the schema validation?

Yes, but i will move them under `config` when i use this version of the cli in the mt-repo.

**Comment by rpriyanshu9** [comment]
> > LGTM. Just one thing, now that the `secrets` are present inside the `config` attribute, won't the existing Addons fail the schema validation?
> 
> Yes, but i will move them under `config` when i use this version of the cli in the mt-repo.

OK then. Looks good to me!

**Review by yashvardhan-kukreja — APPROVED** [review]

**Pull request merged**